mPower2/MotorControl/StepObjects/MCTHandSelectionStepObject.swift

```diff
@@ -0,0 +1,116 @@
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import Foundation
+
+/// A Subclass of RSDFormUIStepObject which uses MCTHandSelectionDataSource.
+public class MCTHandSelectionStepObject : RSDFormUIStepObject {
+    override public func instantiateDataSource(with taskPath: RSDTaskPath, for supportedHints: Set<RSDFormUIHint>) -> RSDTableDataSource? {
+        return MCTHandSelectionDataSource(step: self, taskPath: taskPath, supportedHints: supportedHints)
+    }
+}
+
+/// An enum for representing the different possible usable hands a user may have.
+public enum MCTHandSelection : String, Codable {
+    case left, right, both
+}
+
+/// The object that serves as the data soruce for an MCTHandSelectionStep
+public class MCTHandSelectionDataSource : RSDFormStepDataSourceObject {
+    
+    /// Key for the randomized hand order in the task result.
+    public static let handOrderKey = "handOrder"
+    
+    /// Key for which hands the user said they could use in the task result.
+    public static let selectionKey = "handSelection"
+    
+    /// Override the initial result to look for the user's previous answer to this quesiton in
+    /// UserDefaults.
+    override open var initialResult : RSDCollectionResult? {
+        /// TODO rkolmos 04/10/2018 fix this to not use string builders
+        let initialResultKey : String = "\(self.taskPath.task!.identifier)_lastHandSelection"
+        let defaults = UserDefaults.standard
+        guard let handSelection = defaults.string(forKey: initialResultKey) else { return nil }
+        var ret = self.instantiateCollectionResult()
+        var answerResult = RSDAnswerResultObject(identifier: MCTHandSelectionDataSource.selectionKey, answerType: .string)
+        answerResult.value = handSelection
+        ret.appendInputResults(with: answerResult)
+        return ret
+    }
+    
+    /// Override populateInitialResults to also write a randomized handOrder result.
+    override open func populateInitialResults() {
+        super.populateInitialResults()
+        _updateHandOrder()
+    }
+    
+    /// Override select answer to write the user's choice to UserDefaults, and to write a randomized handOrder result.
+    override open func selectAnswer(item: RSDChoiceTableItem, at indexPath: IndexPath) throws -> (isSelected: Bool, reloadSection: Bool) {
+        let ret = try super.selectAnswer(item: item, at: indexPath)
+        guard let handSelection = _updateHandOrder()
+               else {
+            return (false, false)
```

Comment:
throw an `assertionFailure()` here b/c you are not expecting this to happen. Also, you can still return `ret`.